File: src/infinicore/ops/erf/erf.cc
#include "infinicore/ops/erf.hpp"

namespace infinicore::op {

common::OpDispatcher<Erf::schema> &Erf::dispatcher() {
    static common::OpDispatcher<Erf::schema> dispatcher_;
    return dispatcher_;
};

void Erf::execute(Tensor output, Tensor input) {
    dispatcher().lookup(context::getDevice().getType())(output, input);
}

Tensor erf(Tensor input) {
    auto output = Tensor::empty(input->shape(), input->dtype(), input->device());
    erf_(output, input);
    return output;
}

void erf_(Tensor output, Tensor input) {
    Erf::execute(output, input);
}

} // namespace infinicore::op

File: python/infinicore/ops/erf.py
from infinicore.lib import _infinicore
from infinicore.tensor import Tensor

# 注意函数名必须是 erf
def erf(input, *, out=None):
    if out is None:
        return Tensor(_infinicore.erf(input._underlying))
    _infinicore.erf_(out._underlying, input._underlying)

    return out